### ISSUE TITLE
Performance vs Boost

### ISSUE THREAD

**Issue by stijn-ucrowds** [eadf/boostvoronoi.rs#2]
Thank you for your work on porting the library over to Rust, it has a lot of potential to simplify voronoi diagram generation in Rust. We've been experimenting using this library to generate the voronoi diagrams for our application, which so far has resulted in identical output to the C++ version. One issue is that this library is roughly 2x slower
```
C++ voronoi:   cells:180844, edges:1084654, vertices:361494, in: 1969ms
Rust voronoi:  cells:180844, edges:1084654, vertices:361494, in: 3922ms
```
Do you have any optimizations in mind? Are there any settings we can tweak?

**Comment by stijn-ucrowds** [comment]
It seems like a lot of time is being spent inserting nodes into a linked list. Here is an excerpt from the profiling flame graph.
![image](https://user-images.githubusercontent.com/82461032/151539903-5731b78f-4911-45e3-a196-03b0b369808b.png)

**Comment by eadf** [comment]
Yes, this is probably because of this "hack" that still needs a proper solution: #1
 It will, however, take some time for me to look into that.
